zuul-guice/build.gradle

```diff
@@ -2,7 +2,7 @@ apply plugin: "java-library"
 
 dependencies {
     implementation project(":zuul-core")
-    api(group: 'com.google.inject', name: 'guice', version: "4.2.3", classifier: "no_aop")
+    api(group: 'com.google.inject', name: 'guice', version: "5.0.1")
```

Comment:
From the release notes, looks like they introduced `-Dguice_bytecode_gen_option=DISABLED` to stick to not using AOP based code. Thoughts?

Reply:
the aop thing was to work around JDK11 problems iirc.  Using the supported byte code reader/generator should obviate the need for aop specific stuff.

Reply:
Right, so probably worth making it explicit setting that flag internally, imo.

Reply:
Well, what I am saying is that the sole reason for the classifier was to work around a bug with how cglib read and generated classes.  Now that cglib is not being used, there is no reason to be for or against codegen.  Those generated classes tend to have performance optimizations, so I'd prefer to have them, as long as they dont cause trouble.    In other words, I don't think we need to set that System property.

Reply:
cglib removal only changes what gets used for bytecode generation. My point being, as of today, `no_aop` implies that we don't intend on supporting aop based interception. This changes that behavior.

I'm not a fan of bytecode generation being a requirement for DI.